MenuApplets/Clock/AnalogClockWidget.cpp

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2020, the SerenityOS developers.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "AnalogClockWidget.h"
+#include <LibGfx/Palette.h>
+#include <LibGfx/Point.h>
+#include <LibGfx/Rect.h>
+#include <math.h>
+
+AnalogClockWidget::~AnalogClockWidget()
+{
+}
+
+void AnalogClockWidget::paint_event(GUI::PaintEvent&)
+{
+    GUI::Painter painter(*this);
+
+    Gfx::IntRect bounds { 0, 0, width(), height() };
+    painter.fill_ellipse(bounds, Color::DarkGray);
+    auto width_thickness = width() / 15;
+    auto height_thickness = height() / 15;
+    bounds.shrink(width_thickness, height_thickness);
+    painter.fill_ellipse(bounds, palette().button());
+
+    painter.fill_ellipse(center_circle, Color::DarkGray);
+
+    painter.draw_text(rect_twelve, StringView("12", 2), Gfx::TextAlignment::Center, Color::DarkGray);
+    painter.draw_text(rect_three, StringView("3", 1), Gfx::TextAlignment::Center, Color::DarkGray);
+    painter.draw_text(rect_six, StringView("6", 1), Gfx::TextAlignment::Center, Color::DarkGray);
+    painter.draw_text(rect_nine, StringView("9", 1), Gfx::TextAlignment::Center, Color::DarkGray);
```

Comment:
This definitely should be `palette().base_text()`, now that it's printed on background that might be anything, for example dark grey.

I'm not sure which colors to use for the ellipse and the hands. I suggest `palette().base_text()`, but many others make sense. You can use the ThemeEditor to browse the colors, or look at the theme `.ini`s.

Reply:
Got it!!